src/components/PrivateRoute/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import {
+  Route,
+  Redirect,
+  RouteComponentProps,
+  RouteProps
+} from "react-router-dom";
+import { useAuthStateContext } from "../../context/auth";
+
+type PrivateRouteProps = RouteProps & {
+  component: React.FC<RouteComponentProps>;
+};
+
+export const PrivateRoute: React.FC<PrivateRouteProps> = ({
+  component: Component,
+  ...rest
+}) => {
+  const { isAuthenticated } = useAuthStateContext();
+
+  const render = props =>
```

Comment:
Declaring the render function for the render prop inside this function will construct a new render function when any prop to `PrivateRoute` changes which could lead to unnecessary renders. We should declare render outside of `PrivateRoute`

Reply:
it needs the `isAuthenticated` value which is only available within the context of the component. Also react is really fast and this will not cause a problem in any capacity we need to be worried about

Reply:
I was recommending that `isAuthenticated` should be passed to the render function as a prop. By doing this we can avoid rerendering all children of the render component in certain cases with a small refactor.

Reply:
these micro optimzations to avoid re-renders, which will happen very seldomly given the nature and location of this component in the react tree, will have an immaterial difference in the performance of this application.